### PR TITLE
Add documentation for String.fromUTF8, numeric types' fromString

### LINKED ISSUE #2056: Documentation for new standard library functions
### Issue To Be Solved

The new standard library functions introduced in #1982 and #1927 need documentation.

### Suggested Solution

- [ ] Document the functions
- [ ] While at it, maybe check if other standard library functions are undocumented and document them

### PR DESCRIPTION
**Pull request by dreamsmasher**
Closes #2056

## Description

New standard library functions were introduced in #1927 and #1982, but ~~i was too lazy to write docs~~ they were omitted from the Cadence language reference. This PR extends our docs to cover the new functionality. 
______

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Comment by turbolent** [comment]
@dreamsmasher The "Chore" label is used for "meta" tasks like updating dependencies, changing CI, syncing branches, etc. This PR is simply adding documentation, so that's enough

File: docs/language/values-and-types.mdx [hunk 1]
```diff
@@ -168,6 +168,20 @@ Integers have multiple built-in functions you can use.
   answer.toString()  // is "42"
   ```
 
+- `cadence•fun fromString(_ input: String): T?`
```
Comment (turbolent) [inline docs/language/values-and-types.mdx:171]: Unlike the other functions documented here, this is not an instance function, but a "static" function on the type itself. Maybe document it as:

```suggestion
- `cadence•fun T.fromString(_ input: String): T?`
```

And document that `T` is any integer type.

Maybe also separate this from the remaining instance functions by moving it to a separate "section" below, e.g.

"All integer types support the following functions:"

File: docs/language/values-and-types.mdx [hunk 2]
```diff
@@ -222,6 +236,21 @@ Fixed-Point numbers have multiple built-in functions you can use.
   fix.toString()  // is "1.23000000"
   ```
 
+- `cadence•fun fromString(_ input: String): T?`
```
Comment (turbolent) [inline docs/language/values-and-types.mdx:239]: Same as above

File: docs/language/values-and-types.mdx [hunk 3]
```diff
@@ -651,6 +680,13 @@ Strings have multiple built-in functions you can use:
   // `bytes` is `[70, 108, 111, 119, 101, 114, 115, 32, 240, 159, 146, 144]`
   ```
 
+- `cadence•fun fromUTF8(_ input: [UInt8]): String?`
```
Comment (turbolent) [inline docs/language/values-and-types.mdx:683]: Same as above

**Review by turbolent — COMMENTED** [review]
Nice! While at it, could you please also add documentation for the new `String.fromCharacters` function?

**Comment by dreamsmasher** [comment]
> @dreamsmasher The "Chore" label is used for "meta" tasks like updating dependencies, changing CI, syncing branches, etc. This PR is simply adding documentation, so that's enough

ah my bad, sorry. writing docs defs feels like a chore at least

**Review by turbolent — APPROVED** [review]
Nice! 👌 Thank you for also adding examples 👏

**Review by SupunS — APPROVED** [review]

**Pull request merged**